crates/quantum_info/src/sparse_pauli_op.rs

```diff
@@ -1414,7 +1417,16 @@ mod tests {
                 .unwrap();
             let expected: DecomposeMinimal = paulis.into();
             let actual: DecomposeMinimal = decompose_dense_inner(arr.view(), 0.0).unwrap().into();
+            #[cfg(not(miri))]
             assert_eq!(actual, expected);
+            #[cfg(miri)]
+            {
+                assert!(actual.coeffs.abs_diff_eq(&expected.coeffs, 1e-8));
+                assert_eq!(actual.z, expected.z);
+                assert_eq!(actual.x, expected.x);
+                assert_eq!(actual.phases, expected.phases);
+                assert_eq!(actual.num_qubits, expected.num_qubits);
+            }
```

Comment:
Why is `DecomposeMinimal`'s `PartialEq` implementation not miri safe?

Reply:
The details are in the commit message that made this change: https://github.com/Qiskit/qiskit/pull/14873/commits/3e4dae30caa19953852c593ef3ea207f4652d662 miri has a purposely low precision float implementation so the float comparisons need to have a larger tolerance to work with miri